### PR TITLE
Stop propagation of widget dom events

### LINKED ISSUE #478: Stop widget mouse / touch / key events propagating
**Enhancement** 

We should use `event.stopPropagation()` to prevent mouse / touch / key events leaking outside of widgets

### PR DESCRIPTION
**Pull request by tomdye**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [ ] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

Resolves #478

### CODE REVIEW

File: src/checkbox/Checkbox.ts [hunk 1]
```diff
@@ -63,13 +63,28 @@ export const ThemedBase = ThemedMixin(WidgetBase);
...
-	private _onTouchEnd (event: TouchEvent) { this.properties.onTouchEnd && this.properties.onTouchEnd(event); }
-	private _onTouchCancel (event: TouchEvent) { this.properties.onTouchCancel && this.properties.onTouchCancel(event); }
+	private _onClick (event: MouseEvent) {
+		event.stopPropagation();
+		this.properties.onClick && this.properties.onClick(event); }
+	private _onFocus (event: FocusEvent) {
+		this.properties.onFocus && this.properties.onFocus(event);
+		this.invalidate();
+	}
+	private _onMouseDown (event: MouseEvent) {
```
Comment (agubler) [inline src/checkbox/Checkbox.ts]: formatting...

File: src/radio/Radio.ts [hunk 2]
```diff
@@ -48,13 +48,28 @@ export class RadioBase<P extends RadioProperties = RadioProperties> extends Them
...
+	private _onClick (event: MouseEvent) {
+		event.stopPropagation();
+		this.properties.onClick && this.properties.onClick(event); }
+	private _onFocus (event: FocusEvent) {
+		this.properties.onFocus && this.properties.onFocus(event);
+		this.invalidate();
+	}
+	private _onMouseDown (event: MouseEvent) {
+		event.stopPropagation();
+		this.properties.onMouseDown && this.properties.onMouseDown(event); }
```
Comment (agubler) [inline src/radio/Radio.ts]: formatting...

File: src/slider/Slider.ts [hunk 3]
```diff
@@ -78,17 +78,35 @@ export class SliderBase<P extends SliderProperties = SliderProperties> extends T
 
 	private _onBlur (event: FocusEvent) { this.properties.onBlur && this.properties.onBlur(event); }
 	private _onChange (event: Event) { this.properties.onChange && this.properties.onChange(event); }
-	private _onClick (event: MouseEvent) { this.properties.onClick && this.properties.onClick(event); }
+	private _onClick (event: MouseEvent) {
+		event.stopPropagation();
+		this.properties.onClick && this.properties.onClick(event); }
```
Comment (agubler) [inline src/slider/Slider.ts]: formatting...

File: src/textarea/Textarea.ts [hunk 4]
```diff
@@ -59,17 +59,35 @@ export const ThemedBase = ThemedMixin(WidgetBase);
 export class TextareaBase<P extends TextareaProperties = TextareaProperties> extends ThemedBase<P, null> {
 	private _onBlur (event: FocusEvent) { this.properties.onBlur && this.properties.onBlur(event); }
 	private _onChange (event: Event) { this.properties.onChange && this.properties.onChange(event); }
-	private _onClick (event: MouseEvent) { this.properties.onClick && this.properties.onClick(event); }
+	private _onClick (event: MouseEvent) {
+		event.stopPropagation();
+		this.properties.onClick && this.properties.onClick(event); }
```
Comment (agubler) [inline src/textarea/Textarea.ts]: formatting...

File: src/textinput/TextInput.ts [hunk 5]
```diff
@@ -70,17 +70,35 @@ export const ThemedBase = ThemedMixin(WidgetBase);
 export class TextInputBase<P extends TextInputProperties = TextInputProperties> extends ThemedBase<P, null> {
 	private _onBlur (event: FocusEvent) { this.properties.onBlur && this.properties.onBlur(event); }
 	private _onChange (event: Event) { this.properties.onChange && this.properties.onChange(event); }
-	private _onClick (event: MouseEvent) { this.properties.onClick && this.properties.onClick(event); }
+	private _onClick (event: MouseEvent) {
+		event.stopPropagation();
+		this.properties.onClick && this.properties.onClick(event); }
```
Comment (agubler) [inline src/textinput/TextInput.ts]: formatting...

File: src/calendar/Calendar.ts [hunk 6]
```diff
@@ -180,6 +180,7 @@ export class CalendarBase<P extends CalendarProperties = CalendarProperties> ext
 	}
 
 	private _onDateKeyDown(event: KeyboardEvent) {
+		event.stopPropagation;
```
Comment (maier49) [inline src/calendar/Calendar.ts]: Should you be calling this?

**Pull request merged**